vtex-apps/store-components
# Disable select options in dropdown

#### What problem is this solving?

Similar to isImpossible prop from SelectorItem component where 'o-20' (opacity) is applied for options when displayMode is default, we should have the ability to disable select options (from dropdown) as well, in the case of displayMode: select

I know hideImpossibleCombinations prop could be used, but the client sometimes wants them displayed but disabled instead of hiding them completely.

#### How to test it?

1. Link this branch on a workspace
2. Link the following PR from vtex.styleguide on the same workspace: https://github.com/vtex/styleguide/pull/1392
3. Add displayMode: "select" and disableUnavailableSelectOptions: true (or use my configuration) for sku-selector and check to see that the dropdown options that are actually "impossible" cases will now appear as disabled instead of enabled.  

`"sku-selector": {
    "props": {
      "visibility": "more-than-one",
      "hideImpossibleCombinations": false,
      "initialSelection": "empty",
      "displayMode": "select",
      "disableUnavailableSelectOptions": true
    }
  }`

#### Screenshots or example usage:

https://user-images.githubusercontent.com/71461884/121042587-81c6ca80-c7bc-11eb-98dc-779ebcd833e7.mp4

#### Describe alternatives you've considered, if any.

hideImpossibleCombinations prop as alternative but the client doesn't want that behaviour so instead I had to go with this solution which is non-intrusive because of the extra prop that is by default set to false and only applies if the displayMode is "select" and if it's specified explicitly by the developer in store-theme (otherwise, in styleguide app, the disabled attribute will default to false and will have no impact)

#### Related to / Depends on

https://github.com/vtex/styleguide/pull/1392

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/7kO9VZjCv3FkI/giphy.gif)
